addons/website/models/ir_http.py

```diff
@@ -275,11 +275,14 @@ def _serve_page(cls):
                 need_to_cache = True
                 try:
                     r = page._get_cache_response(cache_key)
-                    if r['time'] + page.cache_time > time.time():
-                        response = werkzeug.Response(r['content'], mimetype=r['contenttype'])
-                        response._cached_template = r['template']
-                        response._cached_page = page
-                        return response
+                    if page._can_be_cached(r['content']):
+                        if r['time'] + page.cache_time > time.time():
+                            response = werkzeug.Response(r['content'], mimetype=r['contenttype'])
+                            response._cached_template = r['template']
+                            response._cached_page = page
+                            return response
+                    else:
+                        need_to_cache = False
```

Comment:
Just to be sure I get this correctly.

We are always adding the page to the cache, event if `_can_be_cached()` is false, as it will break outside of the try/except before reaching the if/else, with `need_to_cache` previously set to `True`?
Then it will work since it will be cached but no one will actually read from the cache, right?
I guess that's the compromise to be able to serve a cached page without having to render it (obviously) to know if it can be cached.

Reply:
Worth comments in the code indeed IMO :open_mouth: